### ISSUE TITLE
[YSQL] Reduce per-connection overhead for cache preloading

### ISSUE THREAD

**Issue by m-iancu** [YugaByte/yugabyte-db#1091]
Currently, running [sample apps](https://github.com/YugaByte/yb-sample-apps) with over `100` connections in a `3`-node cluster is bottlenecked by all connections hitting the yb-master to preload their local caches. 
Same issue can be observed on a local cluster when using enough connections. 

To replicate:
1. Start cluster
```
./bin/yb-ctl destroy; ./bin/yb-ctl create
```
2. Run sample apps with large number of threads e.g. 99 (should see no ops for quite a while)
```
java -jar ./target/yb-sample-apps.jar --workload SqlInserts --nodes 127.0.0.1:5433 --num_threads_write 99 --num_threads_read 0 --num_unique_keys 1000000000
```
```
0 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
26 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Using a randomly generated UUID : 4d852c5d-af09-44cf-b718-af6b834169d7
37 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - App: SqlInserts
37 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Run time (seconds): -1
37 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Adding node: 127.0.0.1:5433
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num reader threads: 0, num writer threads: 99
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num unique keys to insert: 1000000000
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to update: -1000000001
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to read: -1
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Value size: 0
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Restrict values to ASCII strings: false
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Perform sanity check at end of app run: false
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Table TTL (secs): -1
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Local reads: false
38 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Read only load: false
1088 [main] INFO com.yugabyte.sample.apps.SqlInserts  - Dropping any table(s) left from previous runs if any
5245 [main] INFO com.yugabyte.sample.apps.SqlInserts  - Created table: postgresqlkeyvalue
10251 [Thread-1] INFO com.yugabyte.sample.common.metrics.MetricsTracker  - Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Uptime: 5006 ms |
15253 [Thread-1] INFO com.yugabyte.sample.common.metrics.MetricsTracker  - Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Uptime: 10008 ms |
20257 [Thread-1] INFO com.yugabyte.sample.common.metrics.MetricsTracker  - Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Uptime: 15012 ms |
25260 [Thread-1] INFO com.yugabyte.sample.common.metrics.MetricsTracker  - Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Uptime: 20015 ms |
```

**Comment by m-iancu** [comment]
Closed in 465b6d6.

**Comment by kmuthukk** [comment]
nice work @m-iancu !
